packages/build-utils/src/file-ref.ts

```diff
@@ -139,3 +142,30 @@ export default class FileRef implements FileBase {
     });
   }
 }
+
+/**
+ * Get the value of the environment variable `key` as a valid URL string or `null` if unset.
+ *
+ * @param key The `process.env` member which holds the value to validate
+ * @returns The valid URL string or `null` if `key` is not set on `process.env`
+ */
+function getEnvAsUrlOrThrow(
+  key: keyof (typeof process)['env']
+): ReturnType<URL['toString']> | null {
+  const value = process.env[key];
+
+  if (value === undefined) return null;
```

Comment:
I originally had `if (!value) return null`, but now that we're checking for explicit `undefined` (to explicitly error on other falsy values) I wonder if `null` still makes sense as a return value?

Reply:
In general: I prefer to use `undefined` over `null` in essentially all cases, but that's me. ([ref](https://massalabs.com/blog/prefer-null-or-undefined-in-javascript))

Here specifically: Because this is a wrapper over `process.env[someKey]`, which returns `undefined` if the key has no value, people will probably assume the return value of this function is also `undefined` in that case.